src/main/java/com/codeborne/selenide/ElementsCollection.java

```diff
@@ -261,6 +261,26 @@ public SelenideElement last() {
     return get(size() - 1);
   }
 
+  /**
+   * returns the first n elements of the collection
+   * @param elements - number of elements
+   * @return
+   */
+  public ElementsCollection first(int elements) {
+    List<WebElement> sublist = getActualElements().subList(0, Math.min(elements, size()));
+    return new ElementsCollection(new WebElementsCollectionWrapper(sublist));
```

Comment:
This implementation is not correct. It's not "reactive" enough.

GIVEN "1", "2", "c", "d" texts in the '.item' list during first 1 second
WHEN texts change to "a", "b", "c", "d" during next 5 seconds
THEN the code `$$(".item").first(2).shouldHave(texts("a", "b"))` still will fail on timeout of 6 seconds
BUT it should pass. 

This test by the way should also be added to the test suite. 

The correct way to implement these methods is the same as for "filterBy" method.

Reply:
I'm not sure this error is caused by my changes. It looks like the .shouldHave(texts()) is not working properly for collections. Using your example, lets say we have one element with text "1" which changes text to "a" after 5 seconds. If you call $$(".item").shouldHave(texts("a")) without my first() method it will still fail after 6 seconds. Even if you set the Configuration.timeout to any other value that line still fails after 6 seconds timeout.

Reply:
It was not about waiting 6 seconds. It's about waiting at most 6 seconds and succeed to find elements if they appear or change. Your implementation is not dynamic, it will not succeed. It will remember the exact number of elements at the very beginning - and will not update them after they change during the period of 6 seconds or whatever is set in the Configuration.collectionTimeout

Nevertheless - write a test and check it:)

Just a few tests is not enough for good test coverage of a new feature ;)

You checked only the "subListing" in context of counting size. 
But selenide is much more - it is also:
- dynamic elements that are updated when needed
- waiting elements
- informative error messages

I believe we should ensure that all those features are supported by the implementation of new first/last methods, and so we should have tests for them